relay-dynamic-config/src/metrics.rs

```diff
@@ -170,13 +171,113 @@ impl TransactionMetricsConfig {
     }
 }
 
+/// Combined view of global and project-specific metrics extraction configs.
+#[derive(Debug, Clone)]
+pub struct CombinedMetricExtractionConfig<'a> {
+    global: &'a MetricExtractionTemplates,
+    project: &'a MetricExtractionConfig,
+}
+
+impl<'a> CombinedMetricExtractionConfig<'a> {
+    /// Creates a new combined view from two references.
+    pub fn new(global: &'a MetricExtractionTemplates, project: &'a MetricExtractionConfig) -> Self {
+        Self { global, project }
+    }
+
+    /// Returns an iterator of metric specs.
+    pub fn metrics(&self) -> impl Iterator<Item = &MetricSpec> {
+        let project = self.project.metrics.iter();
+        let enabled_global = self
+            .enabled_templates()
+            .flat_map(|template| template.metrics.iter());
+
+        project.chain(enabled_global)
+    }
+
+    /// Returns an iterator of tag mappings.
+    pub fn tags(&self) -> impl Iterator<Item = &TagMapping> {
+        let project = self.project.tags.iter();
+        let enabled_global = self
+            .enabled_templates()
+            .flat_map(|template| template.tags.iter());
+
+        project.chain(enabled_global)
+    }
+
+    fn enabled_templates(&self) -> impl Iterator<Item = &MetricExtractionTemplate> {
+        self.global.templates.iter().filter_map(|(key, template)| {
+            let is_enabled_by_override =
+                self.project.global_templates.get(key).map(|c| c.is_enabled);
+            let is_enabled = is_enabled_by_override.unwrap_or(template.is_enabled);
```

Comment:
What should Relay do when two metrics have different tags in different templates? For example, the project config sets `key1:value1` in `template1`, but the global config sets `key2:value2` in `template2`.

Reply:
IMO this is worth documenting, it's easy to assume that tags are incremental.